container/endpoint.go

```diff
@@ -692,15 +703,52 @@ func (c *Controller) registerExportedEndpoints() error {
 
 				// TODO: avoid set if item already exist with data we want
 				var path string
-				if path, err = vhostRegistry.SetItem(conn, vhost, vhostEndpoint); err != nil {
+				if path, err = publicEndpointRegistry.SetItem(conn, pepKey, vhostEndpoint); err != nil {
 					glog.Errorf("could not register vhost %s for %s: %v", vhost, epName, err)
 					return err
 				} else {
 					glog.Infof("Registered vhost %s for %s at %s", vhost, epName, path)
-					c.vhostZKPaths = append(c.vhostZKPaths, path)
+					c.publicEndpointZKPaths = append(c.publicEndpointZKPaths, path)
 				}
+			}
+
+			//register ports
+			for _, port := range export.ports {
+				portStr := fmt.Sprintf("%d", port)
+				glog.V(1).Infof("registerExportedEndpoints: vhost epName=%s", epName)
+				//delete any existing vhost that hasn't been cleaned up
+				vhostEndpoint := registry.NewPublicEndpoint(epName, endpoint)
+				pepKey := registry.GetPublicEndpointKey(portStr, registry.EPTypePort)
 
+				if paths, err := publicEndpointRegistry.GetChildren(conn, pepKey); err != nil {
+					glog.Errorf("error trying to get previous vhosts: %s", err)
+				} else {
+					glog.V(1).Infof("cleaning vhost paths %v", paths)
+					//clean paths
+					for _, path := range paths {
+						if vep, err := publicEndpointRegistry.GetItem(conn, path); err != nil {
+							glog.V(1).Infof("Could not read %s", path)
+						} else {
+							glog.V(4).Infof("checking instance id of %#v equal %v", vep, c.options.Service.InstanceID)
+							if strconv.Itoa(vep.InstanceID) == c.options.Service.InstanceID {
+								glog.V(1).Infof("Deleting stale port registration for %v at %v ", portStr, path)
+								conn.Delete(path)
+							}
+						}
+					}
+				}
+
+				// TODO: avoid set if item already exist with data we want
```

Comment:
Yes, the TODO is still applicable.  If an existing port is modified, then we do not check for equality and skip the update.  We remove the old node and replace it with an identical one.